servers/interface.go

```diff
@@ -0,0 +1,12 @@
+package servers
+
+import (
+	"github.com/roadrunner-server/http/v4/common"
+)
+
+// internal interface to start-stop http servers
+type InternalServer[T any] interface {
```

Comment:
Generics in Go are not powerful :(((

Reply:
I tried to go more into union types, but it was even worse